includes/class-wc-referralcandy-integration.php

```diff
@@ -73,67 +87,73 @@ public function sanitize_settings($settings) {
-
-            // make sure first name is always populated to avoid checksum errors
-            if (empty(strip_tags($billing_first_name)) === true) { // if first name is empty
-                // extract name from email (i.e. john from john+doe@domain.com or john_doe from john_doe@domain.com)
-                preg_match('/(?<extracted_name>\w+)/', $billing_email, $matches);
-                $billing_first_name = $matches['extracted_name']; // assign extracted name as first name
             }
+        }
 
-            $divData = [
-                'id'                => $this->is_option_enabled('popup')? 'refcandy-popsicle' : 'refcandy-mint',
-                'data-app-id'       => $this->get_option('app_id'),
-                'data-fname'        => $billing_first_name,
-                'data-lname'        => $billing_last_name,
-                'data-email'        => $this->is_option_enabled('popup')? $billing_email : $encoded_email,
-                'data-amount'       => $order_total,
-                'data-currency'     => $order_currency,
-                'data-timestamp'    => $timestamp,
-                'data-external-reference-id' => $order_number,
-                'data-signature'    => md5($billing_email.','.$billing_first_name.','.$order_total.','.$timestamp.','.$this->get_option('secret_key'))
-            ];
+        public function rc_submit_purchase($order_id) {
+            $rc_order = new RC_Order($order_id);
+            $rc_order->submit_purchase();
+        }
+
+        public function rc_process_return($order_id) {
+            $rc_order = new RC_Order($order_id);
+            $rc_order->process_return();
+        }
+
+        public function render_post_purchase_popup($order_id) {
+            $rc_order = new RC_Order($order_id);
 
-            $popsicle_script = '<script>(function(e){var t,n,r,i,s,o,u,a,f,l,c,h,p,d,v;z="script";l="refcandy-purchase-js";c="refcandy-popsicle";p="go.referralcandy.com/purchase/";t="data-app-id";r={email:"a",fname:"b",lname:"c",amount:"d",currency:"e","accepts-marketing":"f",timestamp:"g","referral-code":"h",locale:"i","external-reference-id":"k",signature:"ab"};i=e.getElementsByTagName(z)[0];s=function(e,t){if(t){return""+e+"="+encodeURIComponent(t)}else{return""}};d=function(e){return""+p+h.getAttribute(t)+".js?lightbox=1&aa=75&"};if(!e.getElementById(l)){h=e.getElementById(c);if(h){o=e.createElement(z);o.id=l;a=function(){var e;e=[];for(n in r){u=r[n];v=h.getAttribute("data-"+n);e.push(s(u,v))}return e}();o.src="//"+d(h.getAttribute(t))+a.join("&");return i.parentNode.insertBefore(o,i)}}})(document);</script>';
+            $div = "<div
+                      id='refcandy-lollipop'
+                      data-id='$rc_order->api_id'
+                      data-fname='$rc_order->first_name'
+                      data-lname='$rc_order->last_name'
+                      data-email='$rc_order->email'
+                      data-accepts-marketing='true'
```

Comment:
Should we default to `false` instead of `true` if we can't get User consent from WooCommerce?